core/services/ocr2/plugins/ocr2keeper/integration_21_test.go

```diff
@@ -230,14 +230,22 @@ func TestIntegration_KeeperPluginLogUpkeep(t *testing.T) {
 
 	t.Run("recover logs", func(t *testing.T) {
 		t.Skip() // TODO: fix test (fails in CI)
+
 		addr, contract := addrs[0], contracts[0]
 		upkeepID := registerUpkeep(t, registry, addr, carrol, steve, backend)
 		backend.Commit()
 		t.Logf("Registered new upkeep %s for address %s", upkeepID.String(), addr.String())
 		// blockBeforeEmits := backend.Blockchain().CurrentBlock().Number.Uint64()
 		// Emit 100 logs in a burst
 		emits := 100
-		emitEvents(testutils.Context(t), t, 100, []*log_upkeep_counter_wrapper.LogUpkeepCounter{contract}, carrol, func() {})
+		i := 0
+		emitEvents(testutils.Context(t), t, 100, []*log_upkeep_counter_wrapper.LogUpkeepCounter{contract}, carrol, func() {
+			i++
+			if i%(emits/4) == 0 {
+				backend.Commit()
+				time.Sleep(time.Millisecond * 250) // otherwise we get "invalid transaction nonce" errors
```

Comment:
i think adding constant time values are going to add to flakiness, but can be resolved in another PR